File: minwage.r
require(ggplot2)

#setwd('~/Desktop/r scripts')

#reads in poverty series and makes index into colnames
df.poverty <- read.csv('data/state2yrpovrate1993-2010.csv',na.strings = 'NA')
rownames(df.poverty) <- df.poverty[,'X']
df.poverty <- df.poverty[colnames(df.poverty)[colnames(df.poverty)!='X']]

#reads in minwage series and makes index into colnames
df.minwage <- read.csv('data/minwage.csv',na.strings = '...',sep = '\t')
rownames(df.minwage) <- df.minwage[,'State.or.other.jurisdiction']
df.minwage <- df.minwage[colnames(df.minwage)[colnames(df.minwage)!='State.or.other.jurisdiction']]

#Fills in NAs with federal minwage
for(col in colnames(df.minwage)){
    df.minwage[col][is.na(df.minwage[col])] <- df.minwage['FLSA',col]
}

#fixes labeling discrepancy
rownames(df.minwage)[rownames(df.minwage) == 'District of Columbia'] <- 'D.C.'
rownames(df.minwage)[rownames(df.minwage) == 'FLSA'] <- 'U.S.'

#constrains columns to same range
colnames(df.poverty) <- seq(1994,2004,1)
colnames(df.minwage) <- c('1988','1991','1992','1994','1996','1997','1998','2000','2001','2002','2003','2004','2005','2006')
df.minwage <- df.minwage[,c('1994','1996','1997','1998','2000','2001','2002','2003')]

#makes column names distinguishable
#colnames(df.poverty) <- paste('p.',colnames(df.poverty))
#colnames(df.minwage) <- paste('mw.',colnames(df.minwage))

#joins two dataframes and removes memory storage of df.minwage
#df.poverty <- cbind(df.poverty,df.minwage)
#rm(df.minwage)


#returns years with different minimum wages from previous year
get.vlines <- function(col){
  #sets up comparison vector
  lagged <- append(df.minwage[col,],NA,after = 0)[1:8]
  return(which(df.minwage[col,] > lagged))

}

makeggplots <- function(locale = 'U.S.') {
#write it from scratch because this isn't working.
  
  vlines <- get.vlines(locale)
  cols <- colnames(df.poverty)
  
  pl <- ggplot(aes(x = cols[1:11],y = df.poverty[locale,1:11]))
  pl <- pl + geom_point()
  
#  ggsave(filename=paste('plots/',locale,'.png'),height=6)
}

File: minwage.py
import pandas as pd
import numpy as np
from ggplot import *

#Poverty rates by state from source in readme
df_poverty = pd.read_csv('data/state2yrpovrate1993-2010reshaped.csv',
                         na_values = 'NA')
#Fixes obnoxious state labeling where states end in periods for some idiotic reason
for a in df_poverty.index:
    if df_poverty.ix[a,'State'][-1:] == '.' and df_poverty.ix[a,'State'] != 'D.C.' and df_poverty.ix[a,'State'] != 'U.S.':
        df_poverty.ix[a,'State'] = df_poverty.ix[a,'State'][:-1]
df_poverty['Value'] = df_poverty['Value']/df_poverty['Value'].mean()

#Minwage dataframe from source in readme
#Use to get vlines dataframe showing when minwage changed by geographic locale
df_minwage = pd.read_csv('data/minwage.csv',
                         na_values = '...')

#replaces all NaNs with FLSA for that year
for year in pd.unique(df_minwage['Year'].ravel()):
    flsa = float(df_minwage['Value'][df_minwage['State or other jurisdiction'] == 'FLSA'][df_minwage['Year'] == year])
    df_minwage['Value'][df_minwage['Year'] == year] = df_minwage['Value'][df_minwage['Year'] == year].replace(np.nan,flsa)


#fixes geographic naming discrepancy between minwage and poverty datasets
df_minwage['State or other jurisdiction'][df_minwage['State or other jurisdiction'] == 'District of Columbia'] = 'D.C.'
df_minwage['State or other jurisdiction'][df_minwage['State or other jurisdiction'] == 'FLSA'] = 'U.S.'
df_minwage.columns = pd.Index(['State','Year','Value', 'Type'])
df_minwage['Value'] = df_minwage['Value']/df_minwage['Value'].mean()

#throws 'em into the same df. Which value a line is is indicated by df['Type']
df = pd.concat([df_poverty,df_minwage])

colors = ['#f43605','#0000FF', 'green'] #I have no idea why this has to be three values considering that there are only minimum wage and poverty rate in df['Type'] but whatever

#make the big ol' plot
pl = ggplot(df, aes('Year','Value', colour = 'factor(Type)')) + \
            scale_colour_manual(values = colors) + \
            geom_line() + \
            facet_wrap('State', scales = 'fixed') + \
            ggtitle('Poverty Rate and Minimum Wage by State') + \
            xlim(1993,2005) + \
            theme(axis_text_x = element_text(angle = 0,
                                             vjust = 0,
                                             hjust = 1,
                                             size = 6),
                         axis_text_y = element_text(size = 6)) + \
            ylab('')
#            xlab('Year') + \
#            ylab('Poverty Rate') + \



ggsave(pl,'plots/minwage_small.png')
ggsave(pl,'plots/minwage_vs_poverty.png',height = 30, width = 40,
       limitsize = False)
